src/main/resources/META-INF/rewrite/java-ee-6.yml

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright 2024 the original author or authors.
+# <p>
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.javaee6
+displayName: Migrate to JavaEE6
+description: These recipes help with the Migration to Java EE 6, flagging and updating deprecated methods.
+tags:
+  - javaee6
+  - deprecated
+recipeList:
+  - org.openrewrite.java.migrate.javaee6.HttpSessionInvalidateMethod
+
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.javaee6.HttpSessionInvalidateMethod
+displayName: Use HttpServletRequest `logout` method for programmatic security logout in Servlet 3.0
+description: Do not rely on HttpSession `invalidate` method for programmatic security logout. Add the HttpServletRequest `logout` method which was introduced in Java EE 6 as part of the Servlet 3.0 specification.
+recipeList:
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: javax.servlet.http.HttpSession invalidate()
+      newMethodName: logout
+      ignoreDefinition: true
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: javax.servlet.http.HttpSession
+      newFullyQualifiedTypeName: javax.servlet.http.HttpServletRequest
+      ignoreDefinition: true
```

Comment:
I'm wondering if this might end up changing too broadly; while the current test works, we are in fact changing _**any reference**_ to `HttpSession` to a reference to `HttpServletRequest`, not just when used to call `invalidate()`.

When comparing these docs pages it appears that `HttpSession` was retained, and perhaps not a direct replacement for `HttpServletRequest` in all cases
https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpSession.html
https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpSession.html
https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpServletRequest.html

Maybe we need to be more selective, and somehow access the active `HttpServletRequest` to call `logout` there.

Reply:
That's a good point. I've been encountering similar requirements for some other recipes I'm looking to create, mainly in needing to filter specific instances. 

I didn't notice anything able to do that in the [basic java recipes](https://docs.openrewrite.org/recipes/java/), and was wondering if I would have to create a new recipe to do so.

Reply:
Yes I feel this most likely would have to be a custom recipe, given how you want to instead call a method on an entirely different type instance, that would need to be accessed from somewhere. Perhaps it would be good to start with a more realistic test, and explore the options from there. For instance I know elsewhere we've inserted comments to tell about necessary migration steps that we can't yet automate; that might be the case here if we can't find a request to call `logout` on.